crates/nu-parser/src/parser.rs

```diff
@@ -613,112 +642,90 @@ pub fn parse_call(
 
     if pos == spans.len() {
         return (
-            Expression::garbage(span(spans)),
+            garbage(span(spans)),
             Some(ParseError::UnknownCommand(spans[0])),
         );
     }
 
-    let name = working_set.get_span_contents(spans[pos]);
-
     let cmd_start = pos;
+    let mut name_spans = vec![];
 
-    if expand_aliases {
-        if let Some(expansion) = working_set.find_alias(name) {
-            let orig_span = spans[pos];
-            //let mut spans = spans.to_vec();
-            let mut new_spans: Vec<Span> = vec![];
-            new_spans.extend(&spans[0..pos]);
-            new_spans.extend(expansion);
-            if spans.len() > pos {
-                new_spans.extend(&spans[(pos + 1)..]);
-            }
-
-            let (result, err) = parse_expression(working_set, &new_spans, false);
-
-            let expression = match result {
-                Expression {
-                    expr: Expr::Call(mut call),
-                    span,
-                    ty,
-                    custom_completion: None,
-                } => {
-                    call.head = orig_span;
-                    Expression {
-                        expr: Expr::Call(call),
-                        span,
-                        ty,
-                        custom_completion: None,
-                    }
-                }
-                x => x,
-            };
+    for word_span in spans[cmd_start..].iter() {
+        // Find the longest group of words that could form a command
+        let bytes = working_set.get_span_contents(*word_span);
 
-            return (expression, err);
+        if is_math_expression_byte(bytes[0]) {
+            break;
         }
-    }
 
-    pos += 1;
+        name_spans.push(*word_span);
 
-    if let Some(mut decl_id) = working_set.find_decl(name) {
-        let mut name = name.to_vec();
-        while pos < spans.len() {
-            // look to see if it's a subcommand
-            let mut new_name = name.to_vec();
-            new_name.push(b' ');
-            new_name.extend(working_set.get_span_contents(spans[pos]));
+        let name = working_set.get_span_contents(span(&name_spans));
```

Comment:
So far, I've been treating `math  sum` the same as `math sum`. With this change, the user will always need to type the latter I think?

Probably good for things to be uniform, but wanted to mention it in case we want the old support.